scaffolder-templates/advanced-workflow/template.yaml

```diff
@@ -107,62 +108,146 @@ spec:
                   default: postgres-password
                 persistencePSQLServiceName:
                   title: PostgreSQL K8s Service Name
                   type: string
                   default: sonataflow-psql-postgresql
-                  description: Name of the service running the PostgreSQL instance.
+                  description: Name of the service running the PostgreSQL instance
                 persistencePSQLServiceNamespace:
                   title: PostgreSQL K8s Service Namespace
                   type: string
                   default: sonataflow-infra
-                  description: Namespace of the service running the PostgreSQL instance.
+                  description: Namespace of the service running the PostgreSQL instance
                 persistencePSQLServicePort:
                   title: PostgreSQL Port
                   type: integer
                   default: 5432
-                  description: Port on which the PostgreSQL instance is running.
+                  description: Port on which the PostgreSQL instance is running
                 persistencePSQLDatabaseName:
                   title: PostgreSQL Database Name
                   type: string
-                  description: Name of the database to use for persistence.
+                  description: Name of the database to use for persistence
                   default: sonataflow
               required:
                 - namespace
                 - argocdNamespace
-                - quayOrgName
-                - quayRepoName
                 - persistencePSQLSecretName
                 - persistencePSQLUserKey
                 - persistencePSQLPasswordKey
                 - persistencePSQLServiceName
                 - persistencePSQLServiceNamespace
                 - persistencePSQLServicePort
                 - persistencePSQLDatabaseName
+    - title: Quay Configuration
+      required:
+        - quayConfig
+      properties:
+        quayConfig:
+          title: Create a new Quay repository with the provided name?
+          type: string
+          default: "Use Existing Repository"
+          enum:
+            - "Use Existing Repository"
+            - "Create New Repository"
+            - "No Quay Config Needed"
+      dependencies:
+        quayConfig:
+          oneOf:
+            - properties:
+                quayConfig:
+                  const: "No Quay Config Needed"
+            - properties:
+                quayConfig:
+                  const: "Use Existing Repository"
+                quayOrgName:
+                  title: Quay Organization Name
+                  type: string
+                  description: The Quay organization name of the published workflow
+                quayRepoName:
+                  title: Quay Repository Name
+                  type: string
+                  description: The Quay repository Name of the published workflow. The repository must exist before deploying the gitops
```

Comment:
those are duplicated with the later dependencies, just show them all the time

In the later state then, only show the needed extended parameters ot just embed `dependencies` into `dependencies` in the same state

Reply:
Duplicating in this way lets me divide the page more nicely to build->quay params. So the Quay name and org fields are after the choice to createquse existing repo, which is more coherent for the user. In addition, this allows me to change the description to match what the user is doing (creating a new repository with the given name, or using an existing).